src/chain.ts

```diff
@@ -158,6 +154,9 @@ export class Chain {
             });
         svg.call(zoom);
         Chain.zoom = zoom;
+        // This group will contain the left and right loaders that display a
+        // spinner when new blocks are being added
+        const gloader = svg.append("g").attr("class", "loader");
```

Comment:
For elements that are unique (in this case we will only have one loader container) the best practice is to use an id instead of a class (that's also a micro-optimization, as retrieving by id is slightly faster than by class). Then, on that kind of element, we can use `d3.Select` instead of `d3.SelectAll`.

Reply:
Thanks for the interesting insight! I modified it 👍